sway-core/src/type_system/type_id.rs

```diff
@@ -31,18 +31,18 @@ impl From<usize> for TypeId {
     }
 }
 
-impl UnresolvedTypeCheck for TypeId {
-    fn check_for_unresolved_types(&self) -> Vec<CompileError> {
+impl CollectTypesMetadata for TypeId {
+    fn collect_types_metadata(&self) -> Vec<TypeMetadata> {
         use TypeInfo::*;
         let span_override = if let TypeInfo::Ref(_, span) = look_up_type_id_raw(*self) {
             Some(span)
         } else {
             None
         };
         match look_up_type_id(*self) {
-            UnknownGeneric { name } => vec![CompileError::UnableToInferGeneric {
-                ty: name.as_str().to_string(),
-                span: span_override.unwrap_or_else(|| name.span()),
+            UnknownGeneric { name } => vec![TypeMetadata::UnresolvedType {
```

Comment:
Since this is after type checking, there will be no further type unification happening -- should we also generate `TypeMetadata::UnresolvedType` if the type is `TypeInfo::Unknown`?

Reply:
Done!

Reply:
Alright, so CI fails because apparently some `Unknown` types linger after type checking 🤔 . I reverted the change for now but I'll create an issue to investigate that weird behavior.

Reply:
Hmmmmmm ok. Thanks for looking into it.